[chromium-base] Build error on x64-windows

Package: chromium-base:x64-windows@86.0.4199.1#5

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.29.30154.0
-    vcpkg-tool version: 2023-12-12-1c9ec1978a6b0c2b39c9e9554a96e3e275f7556e
    vcpkg-scripts version: f7423ee18 2024-06-14 (3 days ago)

**To Reproduce**

`vcpkg install chromium-base`
**Failure logs**

```
CMake Warning at scripts/ports.cmake:167 (message):
  Unexpected READ_ACCESS on variable WIN32 in script mode.

  This variable name insufficiently expresses whether it refers to the target
  system or to the host system.  Use a prefixed variable instead.

  - Variables providing information about the host:

    CMAKE_HOST_<SYSTEM>
    VCPKG_HOST_IS_<SYSTEM>

  - Variables providing information about the target:

    VCPKG_TARGET_IS_<SYSTEM>
    VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)

Call Stack (most recent call first):
  ports/chromium-base/portfile.cmake:2147483647 (z_vcpkg_warn_ambiguous_system_variables)
  ports/chromium-base/portfile.cmake:71 (if)
  scripts/ports.cmake:191 (include)


CMake Warning at scripts/ports.cmake:167 (message):
  Unexpected READ_ACCESS on variable WIN32 in script mode.

  This variable name insufficiently expresses whether it refers to the target
  system or to the host system.  Use a prefixed variable instead.

  - Variables providing information about the host:

    CMAKE_HOST_<SYSTEM>
    VCPKG_HOST_IS_<SYSTEM>

  - Variables providing information about the target:

    VCPKG_TARGET_IS_<SYSTEM>
    VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)

Call Stack (most recent call first):
  ports/chromium-base/portfile.cmake:2147483647 (z_vcpkg_warn_ambiguous_system_variables)
  ports/chromium-base/portfile.cmake:91 (if)
  scripts/ports.cmake:191 (include)


CMake Warning at scripts/ports.cmake:167 (message):
  Unexpected UNKNOWN_READ_ACCESS on variable APPLE in script mode.

  This variable name insufficiently expresses whether it refers to the target
  system or to the host system.  Use a prefixed variable instead.

  - Variables providing information about the host:

    CMAKE_HOST_<SYSTEM>
    VCPKG_HOST_IS_<SYSTEM>

  - Variables providing information about the target:

    VCPKG_TARGET_IS_<SYSTEM>
    VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)

Call Stack (most recent call first):
  ports/chromium-base/portfile.cmake:2147483647 (z_vcpkg_warn_ambiguous_system_variables)
  ports/chromium-base/portfile.cmake:91 (if)
  scripts/ports.cmake:191 (include)


CMake Warning at scripts/ports.cmake:167 (message):
  Unexpected UNKNOWN_READ_ACCESS on variable APPLE in script mode.

  This variable name insufficiently expresses whether it refers to the target
  system or to the host system.  Use a prefixed variable instead.

  - Variables providing information about the host:

    CMAKE_HOST_<SYSTEM>
    VCPKG_HOST_IS_<SYSTEM>

  - Variables providing information about the target:

    VCPKG_TARGET_IS_<SYSTEM>
    VCPKG_DETECTED_<VARIABLE> (using vcpkg_cmake_get_vars)

Call Stack (most recent call first):
  ports/chromium-base/portfile.cmake:2147483647 (z_vcpkg_warn_ambiguous_system_variables)
  ports/chromium-base/portfile.cmake:104 (if)
  scripts/ports.cmake:191 (include)


-- Generating build (x64-windows-dbg)...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: F:/Downloads/curlvcpkg/vcpkg/installed/x64-windows/tools/gn/gn.exe gen F:/Downloads/curlvcpkg/vcpkg/buildtrees/chromium-base/x64-windows-dbg "--args=     use_custom_libcxx=false     clang_use_chrome_plugins=false     forbid_non_component_debug_builds=false     treat_warnings_as_errors=false use_lld=false clang_base_path=\"F:/Downloads/curlvcpkg/vcpkg/downloads/tools/clang/clang-15.0.6\" is_component_build=true is_debug=true symbol_level=2"
    Working Directory: F:/Downloads/curlvcpkg/vcpkg/buildtrees/chromium-base/src/25ce732
    Error code: 1
    See logs for more information:
      F:\Downloads\curlvcpkg\vcpkg\buildtrees\chromium-base\generate-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-gn/vcpkg_gn_configure.cmake:10 (vcpkg_execute_required_process)
  installed/x64-windows/share/vcpkg-gn/vcpkg_gn_configure.cmake:34 (z_vcpkg_gn_configure_generate)
  ports/chromium-base/portfile.cmake:113 (vcpkg_gn_configure)
  scripts/ports.cmake:191 (include)



```
<details><summary>F:\Downloads\curlvcpkg\vcpkg\buildtrees\chromium-base\generate-x64-windows-dbg-out.log</summary>

```
ERROR at //build/config/BUILDCONFIG.gn:401:1: Unknown function.
set_sources_assignment_filter(sources_assignment_filter)
^----------------------------
```
</details>

It looks like gn failed is the actual failure and the spew of warnings above it are unrelated:

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: F:/Downloads/curlvcpkg/vcpkg/installed/x64-windows/tools/gn/gn.exe gen F:/Downloads/curlvcpkg/vcpkg/buildtrees/chromium-base/x64-windows-dbg "--args=     use_custom_libcxx=false     clang_use_chrome_plugins=false     forbid_non_component_debug_builds=false     treat_warnings_as_errors=false use_lld=false clang_base_path=\"F:/Downloads/curlvcpkg/vcpkg/downloads/tools/clang/clang-15.0.6\" is_component_build=true is_debug=true symbol_level=2"
    Working Directory: F:/Downloads/curlvcpkg/vcpkg/buildtrees/chromium-base/src/25ce732
    Error code: 1
    See logs for more information:
      F:\Downloads\curlvcpkg\vcpkg\buildtrees\chromium-base\generate-x64-windows-dbg-out.log
```

gn.exe does not care about this new warning from dg0yt

--- a/ports/chromium-base/portfile.cmake
+++ b/ports/chromium-base/portfile.cmake
@@ -68,7 +68,7 @@ set(OPTIONS "\
     treat_warnings_as_errors=false")
 set(DEFINITIONS "")
 
-if(WIN32)
+if(VCPKG_TARGET_IS_WINDOWS)
     # Windows 10 SDK >= (10.0.19041.0) is required
     # https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk
     SET(VCPKG_POLICY_SKIP_ARCHITECTURE_CHECK enabled)
@@ -76,7 +76,7 @@ if(WIN32)
     set(OPTIONS "${OPTIONS} use_lld=false")
 endif()
 
-if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL Linux)
+if(VCPKG_TARGET_IS_LINUX)
     set(OPTIONS "${OPTIONS} use_allocator=\"none\" use_sysroot=false use_glib=false")
 endif()
 
@@ -88,8 +88,8 @@ if(CLANG MATCHES "-NOTFOUND")
 endif()
 get_filename_component(CLANG "${CLANG}" DIRECTORY)
 get_filename_component(CLANG "${CLANG}" DIRECTORY)
-if((WIN32 AND NOT EXISTS "${CLANG}/bin/clang-cl.exe") OR
-   (APPLE AND NOT EXISTS "${CLANG}/bin/clang"))
+if((VCPKG_TARGET_IS_WINDOWS AND NOT EXISTS "${CLANG}/bin/clang-cl.exe") OR
+   (VCPKG_TARGET_IS_OSX AND NOT EXISTS "${CLANG}/bin/clang"))
     message(FATAL_ERROR "Clang needs to be inside a bin directory.")
 endif()
 set(OPTIONS "${OPTIONS} clang_base_path=\"${CLANG}\"")
@@ -101,7 +101,7 @@ else()
     set(OPTIONS "${OPTIONS} is_component_build=false")
 endif()
 
-if(APPLE)
+if(VCPKG_TARGET_IS_OSX)
     set(OPTIONS "${OPTIONS} enable_dsyms=true")
 endif()
 
@@ -126,7 +126,7 @@ set(TARGETS
     base/third_party/dynamic_annotations
     base/third_party/double_conversion)
 
-if(WIN32)
+if(VCPKG_TARGET_IS_WINDOWS)
     list(APPEND TARGETS base/win:pe_image)
 endif()
 

--- a/ports/chromium-base/vcpkg.json
+++ b/ports/chromium-base/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "chromium-base",
   "version": "86.0.4199.1",
-  "port-version": 5,
+  "port-version": 6,
   "description": "Chromium is an open-source browser project that aims to build a safer, faster, and more stable way for all users to experience the web.",
   "homepage": "https://chromium.googlesource.com/chromium/src",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1614,7 +1614,7 @@
     },
     "chromium-base": {
       "baseline": "86.0.4199.1",
-      "port-version": 5
+      "port-version": 6
     },
     "chronoengine": {
       "baseline": "8.0.0",

--- a/versions/c-/chromium-base.json
+++ b/versions/c-/chromium-base.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7f3a0b54287d0d86a331bc8fdc004086b992ab05",
+      "version": "86.0.4199.1",
+      "port-version": 6
+    },
+    {
       "git-tree": "08b399ae573db73407ae4095c79628553416a992",
       "version": "86.0.4199.1",
       "port-version": 5

